instrumentation/openfeign-9.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/openfeign/OpenFeignInstrumentationSingletons.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.openfeign;
+
+import feign.Response;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.rpc.RpcClientAttributesExtractor;
+
+public class OpenFeignInstrumentationSingletons {
+
+  private OpenFeignInstrumentationSingletons() {}
+
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.openfeign-9.2";
+  private static final Instrumenter<ExecuteAndDecodeRequest, Response> INSTRUMENTER;
+
+  static {
+    INSTRUMENTER =
+        Instrumenter.<ExecuteAndDecodeRequest, Response>builder(
+                GlobalOpenTelemetry.get(), INSTRUMENTATION_NAME, new OpenFeignSpanNameExtractor())
+            .addAttributesExtractor(
```

Comment:
I don't think we have any instrumentations that would use rpc client and http client at the same time. It does raise a question whether such combination can satisfy spec requirements or whether it make sense.

Reply:
> I don't think we have any instrumentations that would use rpc client and http client at the same time. It does raise a question whether such combination can satisfy spec requirements or whether it make sense.

I prefer to regard openfeign as a pseudo-RPC component. Therefore, I think the span name should be different from the HTTP Client component. It needs to reflect both HTTP actions and RPC-related attributes such as downstream services name.
, we have to think that this is indeed a rather special component.